### DISCUSSION TITLE
Enforcing time order when running multiple Dynamic Partitions at once

### DISCUSSION THREAD

**Discussion by Zeulni** [dagster-io/dagster#18628] [Q&A]
Hello everyone, I have the following issue:

We are using Dynamic Partitions to materialize our DBT assets. The partition keys have a time format, but are strings (i.e. Dagster does not know the timing). We are not using Daily Partitions, since the time intervals are unpredictable. 5 partitions for one asset could look like this:

1. '2023-12-04T10:00:00.000+01:00'
2. '2023-12-05T10:00:00.000+01:00'
3. '2023-12-06T10:00:00.000+01:00'
4. '2023-12-07T10:00:00.000+01:00'
5. '2023-12-08T10:00:00.000+01:00'

If several partitions are to be materialized at the same time, it is very important for us that they are materialized in the correct order. For example, if the user executes a backfill for partitions 3 and 4, number 3 should be materialized before number 4 (as the partition key indicates that it is a partition that is one day before number 4).

**Here my 2 questions:**

1. For DBT Assets: How can I enforce a time order when running multiple Dynamic Partitions at once (based on the timestamp in the partition key), e.g. when running a backfill with multiple partitions for one asset? 
2. Is it also possible to raise an error/warning, if one partition is skipped (e.g. when selecting only 3 and 5 for a backfill, but not 4)?

Thanks for your support! :)

**Comment by sryza** [comment]
Hey @Zeulni - there isn't currently a way to do this that I'm aware of. I've filed a corresponding issue to track this request: https://github.com/dagster-io/dagster/issues/18773.
